### PR TITLE
Refactor path handling

### PR DESCRIPTION
**Pull request by mweinelt**
- Use pathlib for path joining sugar using slashes
- Rename BASE_DIR to MEDIA_DIR, which is more meaningful
- Introduce INSTALL_DIR for the installation base dir
- Replace glob.glob usage with no wildcard to file existance check

### CODE REVIEW

File: frigate/test/test_http.py [hunk 1]
```diff
@@ -76,19 +77,19 @@ def setUp(self):
...
                     },
-                    "/media/frigate/clips": {
+                    os.path.join(BASE_DIR, "clips"): {
                         "free": 42429.9,
                         "mount_type": "ext4",
                         "total": 244529.7,
                         "used": 189607.0,
                     },
-                    "/media/frigate/recordings": {
+                    os.path.join(BASE_DIR, "recordings"): {
```
Comment (NickM-27) [inline frigate/test/test_http.py:86]: probably should just use RECORDINGS_DIR here

File: frigate/test/test_http.py [hunk 2]
```diff
@@ -76,19 +77,19 @@ def setUp(self):
                         "total": 67.1,
                         "used": 16.6,
                     },
-                    "/media/frigate/clips": {
+                    os.path.join(BASE_DIR, "clips"): {
```
Comment (NickM-27) [inline frigate/test/test_http.py:80]: probably should just use CLIPS_DIR here

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Comment by mweinelt** [comment]
WDYT about renaming the `BASE_DIR` to `STATE_DIR`? That is what systemd calls the service home dir, where it can store its state. `BASE_DIR` sounds comparatively meaningless.

**Review by NickM-27 — APPROVED** [review]

**Comment by mweinelt** [comment]
Did a few more things now, updated the top post.

File: frigate/http.py [hunk 3]
```diff
@@ -1647,11 +1648,11 @@ def latest_frame(camera_name):
...
         ):
             if current_app.camera_error_image is None:
-                error_image = glob.glob("/opt/frigate/frigate/images/camera-error.jpg")
+                error_image = INSTALL_DIR / "frigate/images/camera-error.jpg"
 
-                if len(error_image) > 0:
+                if error_image.is_file():
                     current_app.camera_error_image = cv2.imread(
-                        error_image[0], cv2.IMREAD_UNCHANGED
+                        error_image, cv2.IMREAD_UNCHANGED
```
Comment (NickM-27) [inline frigate/http.py:1655]: This doesn't work in my testing `TypeError: Can't convert object to 'str' for 'filename'`

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/http.py (hunk 3 above)
Comment (mweinelt) [reply frigate/http.py:1655]: Yeah, will need a type cast. Also taking a look at the typing checks.

File: frigate/app.py [hunk 4]
```diff
@@ -109,10 +109,10 @@ def init_logger(self) -> None:
         root_configurer(self.log_queue)
 
     def init_config(self) -> None:
-        config_file = os.environ.get("CONFIG_FILE", "/config/config.yml")
+        config_file = os.environ.get("CONFIG_FILE", CONFIG_DIR / "config.yml")
```
Comment (NickM-27) [inline frigate/app.py:112]:
```suggestion
        config_file = os.environ.get("CONFIG_FILE", str(CONFIG_DIR / "config.yml"))
```

File: frigate/app.py [hunk 5]
```diff
@@ -109,10 +109,10 @@ def init_logger(self) -> None:
...
 
         # Check if we can use .yaml instead of .yml
-        config_file_yaml = config_file.replace(".yml", ".yaml")
+        config_file_yaml = str(config_file).replace(".yml", ".yaml")
```
Comment (NickM-27) [inline frigate/app.py:115]:
```suggestion
        config_file_yaml = config_file.replace(".yml", ".yaml")
```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/const.py [hunk 6]
```diff
@@ -1,14 +1,23 @@
...
-EXPORT_DIR = f"{BASE_DIR}/exports"
-BIRDSEYE_PIPE = "/tmp/cache/birdseye"
-CACHE_DIR = "/tmp/cache"
+from pathlib import Path
+
+INSTALL_DIR = Path("/opt/frigate")
+
+CONFIG_DIR = Path("/config")
+DEFAULT_DB_PATH = CONFIG_DIR / "frigate.db"
+MODEL_CACHE_DIR = CONFIG_DIR / "model_cache"
```
Comment (mweinelt) [inline frigate/const.py:7]: I feel like this should be below CACHE_DIR, wdyt?

Comment (NickM-27) [reply frigate/const.py:7]: if you are trying to go for alphabetical then cache should probably go at the top, but I think it is fine as is tbh

Comment (mweinelt) [reply frigate/const.py:7]: No, I meant putting the model cache below `/tmp/cache`. But I can also sort the vars.

Comment (NickM-27) [reply frigate/const.py:8]: I don't think that makes sense because it has nothing to do with /cache

Comment (NickM-27) [reply frigate/const.py:8]: let's just leave as is

**Review by NickM-27 — APPROVED** [review]

**Comment by xconverge** [comment]
FYI this is probably caused by this commit, not too sure didn't look into it yet

```
2024-02-05 22:31:59.671188335  Exception in thread frigate_stats_emitter:
2024-02-05 22:31:59.671191877  Traceback (most recent call last):
2024-02-05 22:31:59.671193059    File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
2024-02-05 22:31:59.673078134      self.run()
2024-02-05 22:31:59.673080629    File "/opt/frigate/frigate/stats.py", line 347, in run
2024-02-05 22:31:59.673169918      self.dispatcher.publish("stats", json.dumps(stats), retain=False)
2024-02-05 22:31:59.673172545    File "/usr/lib/python3.9/json/__init__.py", line 231, in dumps
2024-02-05 22:31:59.673436728      return _default_encoder.encode(obj)
2024-02-05 22:31:59.673438778    File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
2024-02-05 22:31:59.673749299      chunks = self.iterencode(o, _one_shot=True)
2024-02-05 22:31:59.673750728    File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
2024-02-05 22:31:59.673815405      return _iterencode(o, 0)
2024-02-05 22:31:59.673831853  TypeError: keys must be str, int, float, bool or None, not PosixPath
2024-02-05 22:32:03.761812773  [2024-02-05 22:32:03] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
2024-02-05 22:32:03.761980439  [2024-02-05 22:32:03] frigate.record.maintainer      ERROR   : startswith first arg must be str or a tuple of str, not PosixPath
2024-02-05 22:32:08.761666261  [2024-02-05 22:32:08] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
2024-02-05 22:32:08.761807327  [2024-02-05 22:32:08] frigate.record.maintainer      ERROR   : startswith first arg must be str or a tuple of str, not PosixPath
2024-02-05 22:32:13.759919422  [2024-02-05 22:32:13] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
2024-02-05 22:32:13.760004250  [2024-02-05 22:32:13] frigate.record.maintainer      ERROR   : startswith first arg must be str or a tuple of str, not PosixPath
2024-02-05 22:32:18.755957725  [2024-02-05 22:32:18] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
2024-02-05 22:32:18.756031574  [2024-02-05 22:32:18] frigate.record.maintainer      ERROR   : startswith first arg must be str or a tuple of str, not PosixPath
2024-02-05 22:32:23.760933789  [2024-02-05 22:32:23] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
2024-02-05 22:32:23.761018100  [2024-02-05 22:32:23] frigate.record.maintainer      ERROR   : startswith first arg must be str or a tuple of str, not PosixPath
2024-02-05 22:32:28.761637842  [2024-02-05 22:32:28] frigate.record.maintainer      ERROR   : Error occurred when attempting to maintain recording cache
```

**Comment by xconverge** [comment]
@mweinelt (sorry for the tag, I forgot in the prev message)

**Comment by lolorc** [comment]
another one ;-)
```
frigate-dev | 2024-02-06 12:09:00.927819889  [2024-02-06 12:09:00] frigate.http                   ERROR   : Exception on /stats [GET]                         
frigate-dev | 2024-02-06 12:09:00.927823475  Traceback (most recent call last):                                                                               
frigate-dev | 2024-02-06 12:09:00.927824810    File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 2190, in wsgi_app                             
frigate-dev | 2024-02-06 12:09:00.927826023      response = self.full_dispatch_request()                                                                      
frigate-dev | 2024-02-06 12:09:00.927827372    File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 1486, in full_dispatch_request                
frigate-dev | 2024-02-06 12:09:00.927828478      rv = self.handle_user_exception(e)                                                                           
frigate-dev | 2024-02-06 12:09:00.927829759    File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 1484, in full_dispatch_request                
frigate-dev | 2024-02-06 12:09:00.927830775      rv = self.dispatch_request()                                                                                 
frigate-dev | 2024-02-06 12:09:00.927832069    File "/usr/local/lib/python3.9/dist-packages/flask/app.py", line 1469, in dispatch_request                     
frigate-dev | 2024-02-06 12:09:00.927833246      return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)                                     
frigate-dev | 2024-02-06 12:09:00.927834379    File "/opt/frigate/frigate/http.py", line 1580, in stats                                                       
frigate-dev | 2024-02-06 12:09:00.927835372      return jsonify(stats)                                                                                        
frigate-dev | 2024-02-06 12:09:00.927836640    File "/usr/local/lib/python3.9/dist-packages/flask/json/__init__.py", line 170, in jsonify                     
frigate-dev | 2024-02-06 12:09:00.927837766      return current_app.json.response(*args, **kwargs)                                                            
frigate-dev | 2024-02-06 12:09:00.927839061    File "/usr/local/lib/python3.9/dist-packages/flask/json/provider.py", line 215, in response
frigate-dev | 2024-02-06 12:09:00.927840209      f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype     
frigate-dev | 2024-02-06 12:09:00.927841494    File "/usr/local/lib/python3.9/dist-packages/flask/json/provider.py", line 180, in dumps
frigate-dev | 2024-02-06 12:09:00.927842470      return json.dumps(obj, **kwargs)                                   
frigate-dev | 2024-02-06 12:09:00.927843615    File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps                                                 frigate-dev | 2024-02-06 12:09:00.927844583      return cls(
frigate-dev | 2024-02-06 12:09:00.927845705    File "/usr/lib/python3.9/json/encoder.py", line 199, in encode                                                 frigate-dev | 2024-02-06 12:09:00.927846821      chunks = self.iterencode(o, _one_shot=True)
frigate-dev | 2024-02-06 12:09:00.927847965    File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode                                             frigate-dev | 2024-02-06 12:09:00.927848976      return _iterencode(o, 0)
frigate-dev | 2024-02-06 12:09:00.927850109  TypeError: '<' not supported between instances of 'str' and 'PosixPath'                                          
```

**Comment by blakeblackshear** [comment]
I deleted this commit from the dev branch

**Pull request merged**